scripts/vet.sh

```diff
@@ -184,4 +184,7 @@ revive -formatter plain ./... >"${REV_OUT}" || true
 # TODO: Remove `|| true` to unskip linter failures once existing issues are fixed.
 (noret_grep -v "unused-parameter" "${REV_OUT}" | not grep -v "\.pb\.go:") || true
 
+# Error if trailing spaces found in any files excluding files in .git directory and *.md files
+find . -path ./.git -prune -o -type f -exec grep -Hn "[[:blank:]]$" {} \; | fail_on_output
```

Comment:
Does `git grep "[[:blank:]]$" | fail_on_output` not work?  That seems a lot simpler.

Reply:
yeah for some reason grep fails `fail_on_output` even when there are no violations of trailing whitespace. We probably need to modify `fail_on_output()` function or use something else

Reply:
I gave it a try on my end, and it seems to be working fine

```console
~/grpc-go pr/purnesh42H/7576*
❯ git grep "[[:blank:]]$" | fail_on_output
~/grpc-go pr/purnesh42H/7576*
❯ echo $?
0

~/grpc-go pr/purnesh42H/7576*
❯ git grep "[[:blank:]]" | fail_on_output
~/grpc-go pr/purnesh42H/7576*
❯ echo $?
1
```

Reply:
but I don't i see Success at the end when running `scripts/vet.sh` with this command even though no violations. That's why I am not sure.

Reply:
See the other places we are doing `git grep` in here then...  E.g.

```sh
# - Ensure that the deprecated protobuf dependency is not used.
not git grep "\"github.com/golang/protobuf/*" -- "*.go" ':(exclude)reflection/test/grpc_testing_not_regenerate/*'
```

Reply:
"git grep '[[:blank:]]$' returns an exit status of 1 when no trailing spaces are found.. Hence, we have to negate it. Let me know if this looks good now